acme/authority.go

```diff
@@ -0,0 +1,236 @@
+package acme
+
+import (
+	"crypto/x509"
+
+	"github.com/pkg/errors"
+	"github.com/smallstep/cli/jose"
+	"github.com/smallstep/nosql"
+)
+
+// ACME is the acme authority interface.
+type ACME interface {
+	GetLink(Link, bool, ...string) string
+	GetDirectory() *Directory
+	NewNonce() (string, error)
+	UseNonce(string) error
+	NewAccount(AccountOptions) (*Account, error)
+	UpdateAccount(string, []string) (*Account, error)
+	DeactivateAccount(string) (*Account, error)
+	GetAccount(string) (*Account, error)
+	GetAccountByKeyID(string) (*Account, error)
+	GetOrdersByAccount(string) ([]string, error)
+	NewOrder(OrderOptions) (*Order, error)
+	GetOrder(string, string) (*Order, error)
+	FinalizeOrder(string, string, *x509.CertificateRequest) (*Order, error)
+	GetAuthz(string, string) (*Authz, error)
+	GetChallenge(string, string) (*Challenge, error)
+	ValidateChallenge(string, string, *jose.JSONWebKey) (*Challenge, error)
+	GetCertificate(string, string) ([]byte, error)
+}
+
+// Authority is the layer that handles all ACME interactions.
+type Authority struct {
```

Comment:
Rename it to ACME

Reply:
Well, authority can be ok as it will be `acme.Authority`, but if you can find a different would be better, to not confuse it with the regular authority.